torch/lib/c10d/FileStore.cpp

```diff
@@ -234,12 +257,17 @@ int64_t FileStore::add(const std::string& key, int64_t i) {
     auto len = value.size();
     ti += std::stoll(std::string(buf, len));
   }
-
+  // Always seek to the end to write
+  file.seek(0, SEEK_END);
   // File cursor is at the end of the file now, and we have an
   // exclusive lock, so we can write the new value.
   file.write(key);
   file.write(std::to_string(ti));
-
+  // Need to update the cache since get will only read the file if the key is
+  // not in the cache, in this case, the key is.
+  auto tiStr = std::to_string(ti);
+  auto newValue = std::vector<uint8_t>(tiStr.begin(), tiStr.end());
+  cache_[key] = std::move(newValue);
```

Comment:
This only fixes the case where you have 1 process doing a get after it did its own add. Once another process does an add, the first process cache entry will be invalid and not updated when calling get. Instead, the expected pattern here is to call add(0) and you will always get the real count.